src/ts/components/subtitleoverlay.ts

```diff
@@ -516,19 +528,80 @@ class SubtitleRegionContainer extends Container<ContainerConfig> {
     }, this.config);
   }
 
-  addLabel(labelToAdd: SubtitleLabel) {
+  private addCueBoxContainer(labelToAdd: SubtitleLabel) {
+    const id = labelToAdd.getConfig().id;
+    this.cueBoxContainers[id] = new SubtitleCueBoxContainer();
+    this.cueBoxContainers[id].addLabel(labelToAdd);
+    setVttCueBoxStyles(this.cueBoxContainers[id], labelToAdd.vtt);
+    this.addComponent(this.cueBoxContainers[id]);
+    this.updateComponents();
+
+    if (labelToAdd.vtt.region && labelToAdd.vtt.region.scroll) {
+      this.getDomElement().scrollTo(0, 9999);
+    }
+  }
+
+  private removeCueBoxContainer(labelToRemove: SubtitleLabel) {
+    const id = labelToRemove.getConfig().id;
+    this.cueBoxContainers[id].removeLabel(labelToRemove);
+    this.removeComponent(this.cueBoxContainers[id]);
+    delete this.cueBoxContainers[id];
+    this.updateComponents();
+  }
+
+  addLabel(labelToAdd: SubtitleLabel, overlaySize?: {width: number, height: number}) {
     this.labelCount++;
+
+    if (labelToAdd.vtt) {
+      if (labelToAdd.vtt.region && overlaySize) {
+        setVttRegionStyles(this, labelToAdd.vtt.region, overlaySize);
+      }
+
+      return this.addCueBoxContainer(labelToAdd);
+    }
+
     this.addComponent(labelToAdd);
     this.updateComponents();
   }
 
   removeLabel(labelToRemove: SubtitleLabel): void {
     this.labelCount--;
+
+    if (labelToRemove.vtt) {
+      return this.removeCueBoxContainer(labelToRemove);
+    }
+
     this.removeComponent(labelToRemove);
     this.updateComponents();
   }
 
   public isEmpty(): boolean {
     return this.labelCount === 0;
   }
-}
\ No newline at end of file
+}
+
+export class SubtitleCueBoxContainer extends Container<ContainerConfig> {
```

Comment:
why is this needed? I think the idea of the `SubtitleRegionContainer` is exactly that.